wire-ios-sync-engine/Source/SessionManager/SessionManager+APIVersionResolver.swift

```diff
@@ -94,24 +94,26 @@ extension SessionManager: APIVersionResolverDelegate {
             self.accountManager.accounts.forEach { account in
 
                 // 1. Tear down the user sessions
-                DispatchQueue.main.sync {
-                    self.tearDownBackgroundSession(for: account.userIdentifier)
-                }
-
-                // 2. Migrate users and conversations
-                CoreDataStack.migrateLocalStorage(
-                    accountIdentifier: account.userIdentifier,
-                    applicationContainer: self.sharedContainerURL,
-                    dispatchGroup: dispatchGroup,
-                    migration: {
-                        try $0.migrateToFederation()
-                    },
-                    completion: { result in
-                        if case .failure = result {
-                            log.error("Failed to migrate account for federation")
-                        }
+                DispatchQueue.main.async {
+                    dispatchGroup.enter()
+                    self.tearDownBackgroundSession(for: account.userIdentifier) {
+                        // 2. Migrate users and conversations
+                        CoreDataStack.migrateLocalStorage(
+                            accountIdentifier: account.userIdentifier,
+                            applicationContainer: self.sharedContainerURL,
+                            dispatchGroup: dispatchGroup,
+                            migration: {
+                                try $0.migrateToFederation()
+                            },
+                            completion: { result in
+                                if case .failure = result {
+                                    log.error("Failed to migrate account for federation")
+                                }
+                            }
+                        )
+                        dispatchGroup.leave()
```

Comment:
don't we leave the group before the migrateLocalStorage is finished here?

Reply:
Don't think so since I only added a new enter/leave pair and didn't touch the existing ones. `CoreDataStack.migrateLocalStorage()` will enter/leave the dispatch group as well.